File: LeetCode/Easy/0566-reshape-the-matrix/0566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        num_of_col = (len(mat) * len(mat[0])) // r
        if num_of_col != c:
            return mat
        
        temp = [col for row in mat for col in row]
        idx = 0
        ret = []
        
        for row in range(r):
            ret.append([])
            for col in range(c):
                ret[-1].append(temp[idx])
                idx += 1

        return ret

File: LeetCode/Easy/0566-reshape-the-matrix/0566-reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        if (mat.size() * mat[0].size() != r * c) return mat;
        
        vector<vector<int>> ret(r, vector<int>(c));
        int row = 0;
        int col = 0;
        
        for (int i = 0; i < mat.size(); ++i){
            for (int j = 0; j < mat[0].size(); ++j) {
                if (col == c) {
                    ++row;
                    col = 0;
                }
                ret[row][col++] = mat[i][j];
            }
        }
        
        return ret;
    }
};